internal/operator-controller/rukpak/convert/helm.go

```diff
@@ -77,3 +94,55 @@ func (r *BundleToHelmChartConverter) ToHelmChart(bundle source.BundleSource, ins
 
 	return chrt, nil
 }
+
+// deriveDefaultWatchNamespace determines the default watch namespace based on bundle's supported install modes
+func (r *BundleToHelmChartConverter) deriveDefaultWatchNamespace(rv1 *bundle.RegistryV1, installNamespace string) string {
+	// Check what install modes the bundle supports
+	supportedModes := make(map[string]bool)
+	for _, installMode := range rv1.CSV.Spec.InstallModes {
+		if installMode.Supported {
+			supportedModes[string(installMode.Type)] = true
+		}
+	}
+
+	// Priority order for defaults:
+	// 1. If supports AllNamespaces -> "" (AllNamespaces)
+	// 2. If supports OwnNamespace -> installNamespace (OwnNamespace)
+	// 3. If supports SingleNamespace -> installNamespace (treat as OwnNamespace since we can't know the target)
+	// 4. Fallback -> "" (AllNamespaces)
+
+	if supportedModes["AllNamespaces"] {
+		return "" // Empty string means AllNamespaces
+	}
+	if supportedModes["OwnNamespace"] {
+		return installNamespace // OwnNamespace
+	}
+	if supportedModes["SingleNamespace"] {
+		return installNamespace // Treat as OwnNamespace since we don't know the target namespace
+	}
+
+	// Fallback to AllNamespaces
+	return ""
```

Comment:
return an error if you get here. There should be no default for SingleNamespace or MultiNamespace. Maybe just say something like config parameter `watchNamespace` required